crates/core_simd/src/vector.rs

```diff
@@ -180,6 +180,35 @@ where
         unsafe { slice.as_ptr().cast::<Self>().read_unaligned() }
     }
 
+    /// Writes a SIMD vector to the first `LANES` elements of a slice.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the slice's length is less than the vector's `Simd::LANES`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # #![feature(portable_simd)]
+    /// # #[cfg(feature = "as_crate")] use core_simd::simd;
+    /// # #[cfg(not(feature = "as_crate"))] use core::simd;
+    /// # use simd::u32x4;
+    /// let mut dest = vec![0; 6];
+    /// let v = u32x4::from_array([1, 2, 3, 4]);
+    /// v.copy_to_slice(&mut dest);
+    /// assert_eq!(&dest, &[1, 2, 3, 4, 0, 0]);
+    /// ```
+    pub fn copy_to_slice(self, slice: &mut [T]) {
+        assert!(
+            slice.len() >= LANES,
+            "slice length must be at least the number of lanes"
+        );
+        // Safety:
+        // - We've checked the length is sufficient
+        // - `T` and `Simd<T, N>` are Copy types.
+        unsafe { slice.as_mut_ptr().cast::<Self>().write_unaligned(self) }
```

Comment:
what happens if the simd type has padding (e.g. for non-power-of-2 types where #319 isn't fixed yet)?

imho we should add an assert that the simd type and corresponding array type have the same size, otherwise `write_unaligned` is permitted to overwrite after-the-end elements with `undef` padding -- unsound. panicking is better than UB and the assert will never fire once #319 is fixed.

Reply:
also, you might want an explicit `Copy` bound since we might add `Simd<&mut _, _>` support -- which isn't Copy.

Reply:
Good point about the padding.  I think I'll leave out the `Copy` bound for because basically every one of `Simd`'s associated functions fail if the element isn't `Copy`, that would probably need a much more significant overhaul.